README.md

```diff
@@ -1,102 +1,189 @@
 # API Security service and filter
 
-This module provides a service and a render filter which can be fed by different API providers to secure access. A centralized
-configuration file provides rules for APIs associating an operation to a required permission.
+This module protects specific API (graphql/rest/views, and others) from unauthorized usage, potential XSS/CSRF attacks 
+and provide support for CORS requests.
 
-## Permission configuration
+It prevents API from being called from anywhere, first in a generic way through a global CORS filter, then per API configuration
 
-API can be restricted based on permissions by adding `org.jahia.modules.api.permissions-*.cfg` files into `digital-factory-data/karaf/etc`.
-These files contain a list of rules which will be checked for each node potentially returned by the API. Rules coming from all files are merged and sorted. Rules can be based on path, node type and/or workspace. 
+It also allows users to specify reduced scopes when using tokens. They can choose what API can be called with a token.
 
-A permission check is always done on a JCR node, and is associated with an API name. Different API names are provided for 
-json views, the JCRest API module, or the GraphQL API.
+The module provides a service which can be used by different API providers to secure access. 
 
-The first rule that matches defines the permission that will be checked on the node to ensure the user is correctly allowed to use the API on it. 
+A centralized configuration file provides rules defining how the APIs can be called.
 
-A permission rule is defined by a list of entries of the following format :
+## Authorization configuration
```

Comment:
Before this section, we should have a detailed section about the order in which requests are processed, even though it might sound obvious.

Explaining that default behavior is to deny access, then explain how the order is made within grants of a single scope and within grants across multiple scopes. And how those work with include and exclude. 
If multiple scopes are attached to a single token, are authorization checked towards those scopes alphabetically or in the order the scopes are detailed in the configuration file. This might have impacts on how the user is perceiving scopes to be applied (for example, scopes in PAT UI are listed alphabetically, but are they executed alphabetically ?).

Reply:
I don't understand what do you mean by order of requests ? Requests are all handled independently.

The default behaviour is explained in this section, "If the request do not hold any scope granting the requested API, access will be denied.". There's no ordering of anything - we don't care which scope will be the "first" to grant as long as one of them is granting the access.

Reply:
If you take this configuration: 

```
graphqlA:
  auto_apply:
  - origin: "hosted"
  grants:
  - node: "none"
    api:
      include: graphql
      exclude: graphql.AdminQuery.version, graphql.JcrNo
  metadata:
    visible: "true"
  description: "Exclude access to version"
  constraints:
  - privileged_user: "true"
graphqlB:
  auto_apply:
  - origin: "hosted"
  grants:
  - node: "none"
    api: "graphql"
  metadata:
    visible: "true"
  description: "Access to all GraphQL nodes"
  constraints:
  - privileged_user: "true"
```

If you attach scope `graphqlA` to a token you're getting permission denied when accessing the version node. If you attach both `graphqlA` and `graphqlB` to a token, then the query returns data (although one of the scopes specify version is denied). It might not make much "sense" to create such a scope, but the point here is to detail how things are process in documentation so that if we have two scopes that have conflicting permissions the user understand what happens.

For example:
 - scope "create site" for some reason specifically exclude access to user mutations
 - scope "create user" 
What happens to user mutations if a token is granted "create site" and "create user".

I hope this helps.

Reply:
A scope cannot "deny" an access. There's only a grant entry - it does grant an access, or does nothing. graphqlB does grant the access , graphqlA does not. I can add something like "If the request holds a scope granting the requested API, access will be granted." to the previous statement.

Reply:
Oh that's much clearer now, for some reason I incorrectly assumed that "exclude" was actually denying some access, while it's actually excluding from what is granted.